waiter/test/waiter/scheduler_test.clj

```diff
@@ -804,3 +804,12 @@
           (is (= expected-state-1 actual-state-1')))
         (testing "applied router state update"
           (is (= expected-state-2 actual-state-2)))))))
+
+(deftest test-scheduler-naming
+  (let [scheduler (reify ServiceScheduler)]
+    (is (nil? (scheduler->name scheduler)))
+    (let [scheduler (attach-name scheduler "foo")]
+      (is (= "foo" (scheduler->name scheduler)))
+      (let [scheduler (attach-name scheduler "bar")]
+        (is (= "bar" (scheduler->name scheduler))))
+      (is (= "foo" (scheduler->name scheduler))))))
```

Comment:
What's the point of these last 3 lines? Why would we ever want to do that? If the name is populated when you call `create-component` then we wouldn't ever see this, right?

Reply:
These tests are there to verify that changing the metadata returns a new object and leaves the previous object untouched.

Reply:
My question was: Why do we care if this is implemented as meta-data or some other way? It doesn't seem like we should care about that.

Reply:
Okay, removed the superfluous assertions.